prov/sockets/src/sock_progress.c

```diff
@@ -285,7 +285,9 @@ static void sock_pe_report_mr_completion(struct sock_domain *domain,
 
 	for (i = 0; i < pe_entry->msg_hdr.dest_iov_len; i++) {
 		fastlock_acquire(&domain->lock);
-		mr = sock_mr_get_entry(domain, pe_entry->pe.rx.rx_iov[i].iov.key);
+		ofi_mr_retrieve(domain->mr_heap, pe_entry->pe.rx.rx_iov[i].iov.len,
```

Comment:
ofi_mr_retrieve seems to be doing more work than necessary here. consider having two functions, one that retrieves mr with key, and one that also validate the range?